libs/ngu/carousel/src/lib/ngu-carousel.directive.ts

```diff
@@ -34,7 +34,7 @@ export class NguCarouselPointDirective {}
   selector: '[nguCarouselDef]'
 })
 export class NguCarouselDefDirective<T> {
-  when: (index: number, nodeData: T) => boolean;
+  when?: (index: number, nodeData: T) => boolean;
```

Comment:
Any reason to make this optional?

Reply:
Ummmm I thought it should be optional because the code is checking if `when` is defined:
```ts
def => def.when && def.when(i, data)
```

Reply:
yeah makes sense, thank you